accurest-core/src/main/groovy/io/codearte/accurest/builder/MethodBodyBuilder.groovy

```diff
@@ -0,0 +1,307 @@
+		bb.startBlock()
+		when(bb)
+		bb.endBlock().addEmptyLine()
+	}
+
+	protected void givenBlock(BlockBuilder bb) {
+		bb.addLine(addCommentSignIfRequired('given:'))
+		bb.startBlock()
+		given(bb)
+		bb.endBlock().addEmptyLine()
+	}
+
+	protected void given(BlockBuilder bb) {
+		bb.addLine(getRequestString())
+		bb.indent()
+		request.headers?.collect { Header header ->
+			bb.addLine(getHeaderString(header))
+		}
+		if (request.body) {
+			bb.addLine(getBodyString(bodyAsString))
+		}
+		if (request.multipart) {
+			multipartParameters?.each { Map.Entry<String, Object> entry -> bb.addLine(getMultipartParameterLine(entry)) }
+		}
+		addColonIfRequired(bb)
+		bb.unindent()
+	}
+
+	protected void when(BlockBuilder bb) {
+		bb.addLine(getResponseString(request))
+		bb.indent()
+
+		String url = buildUrl(request)
+		String method = request.method.serverValue.toString().toLowerCase()
+
+		bb.addLine(/.${method}("$url")/)
+		addColonIfRequired(bb)
+		bb.unindent()
+	}
+
+	protected void then(BlockBuilder bb) {
+		validateResponseCodeBlock(bb)
+		if (response.headers) {
+			validateResponseHeadersBlock(bb)
+		}
+		if (response.body) {
+			bb.endBlock()
+			bb.addLine(addCommentSignIfRequired('and:')).startBlock()
+			validateResponseBodyBlock(bb)
+		}
+	}
+
+	private void validateResponseBodyBlock(BlockBuilder bb) {
+		def responseBody = response.body.serverValue
+		ContentType contentType = getResponseContentType()
+		if (responseBody instanceof GString) {
+			responseBody = extractValue(responseBody, contentType, { DslProperty dslProperty -> dslProperty.serverValue })
+		}
+		if (contentType == ContentType.JSON) {
+			appendJsonPath(bb, getResponseAsString())
+			JsonPaths jsonPaths = JsonToJsonPathsConverter.transformToJsonPathWithTestsSideValues(responseBody)
+			jsonPaths.each {
+				bb.addLine("assertThatJson(parsedJson)" + it.method())
+				addColonIfRequired(bb)
+			}
+			processBodyElement(bb, "", responseBody)
+		} else if (contentType == ContentType.XML) {
+			bb.addLine(getParsedXmlResponseBodyString(getResponseAsString()))
+			addColonIfRequired(bb)
+			// TODO xml validation
+		} else {
+			bb.addLine(getSimpleResponseBodyString(getResponseAsString()))
+			processText(bb, "", responseBody as String)
+			addColonIfRequired(bb)
+		}
+	}
+
+	private ContentType getResponseContentType() {
+		ContentType contentType = recognizeContentTypeFromHeader(response.headers)
+		if (contentType == ContentType.UNKNOWN) {
+			contentType = recognizeContentTypeFromContent(response.body.serverValue)
+		}
+		return contentType
+	}
+
+	protected void appendJsonPath(BlockBuilder blockBuilder, String json) {
+		blockBuilder.addLine(("DocumentContext parsedJson = JsonPath.parse($json)"))
+		addColonIfRequired(blockBuilder)
+	}
+
+	protected void processText(BlockBuilder blockBuilder, String property, String value) {
+		if (value.startsWith('$')) {
+			value = value.substring(1).replaceAll('\\$value', "responseBody$property")
```

Comment:
It looks really unreadable -maybe extract it to a utility method `stripFirstChar` if it is not exising already in some StringUtils